src/sentry/runner/commands/backup.py

```diff
@@ -525,6 +531,89 @@ def encrypt(
             dest.write(encrypted.getbuffer())
 
 
+@backup.command(name="sanitize")
+@click.argument("dest", type=click.File("wb"))
+@click.option(
+    "--decrypt-with",
+    type=click.File("rb"),
+    help=DECRYPT_WITH_HELP,
+)
+@click.option(
+    "--decrypt-with-gcp-kms",
+    type=click.File("rb"),
+    help=DECRYPT_WITH_GCP_KMS_HELP,
+)
+@click.option(
+    "--encrypt-with",
+    type=click.File("rb"),
+    help=ENCRYPT_WITH_HELP,
+)
+@click.option(
+    "--encrypt-with-gcp-kms",
+    type=click.File("rb"),
+    help=ENCRYPT_WITH_GCP_KMS_HELP,
+)
+@click.option(
+    "--days-offset",
+    type=int,
+    help="The number of days to adjust the date range seen in the JSON being sanitized.",
+)
+@click.option(
+    "--src",
+    required=True,
+    type=click.File("rb"),
+    help="The input JSON file that needs to be sanitized.",
+)
+@configuration
+def sanitize_(
+    dest: IO[bytes],
+    decrypt_with: IO[bytes],
+    decrypt_with_gcp_kms: IO[bytes],
+    encrypt_with: IO[bytes],
+    encrypt_with_gcp_kms: IO[bytes],
+    days_offset: int | None,
+    src: IO[bytes],
+) -> None:
+    """
+    Sanitize PII from a backup.
+    """
+
+    decryptor = get_decryptor_from_flags(decrypt_with, decrypt_with_gcp_kms)
+
+    # Decrypt the tarball, if the user has indicated that this is one via the use of one of the
+    # `--decrypt...` flags.
+    if decryptor is not None:
+        try:
+            input: IO[bytes] = BytesIO(decrypt_encrypted_tarball(src, decryptor))
+        except DecryptionError as e:
+            click.echo(f"Invalid tarball: {str(e)}", err=True)
+            raise
+    else:
+        input = src
+
+    # Now read the input string into memory as JSONData.
+    try:
+        unsanitized_json = json.load(input)
+    except json.JSONDecodeError:
+        click.echo("Invalid JSON", err=True)
+        raise
+
+    # Perform the sanitization.
+    datetime_offset = timedelta(days=days_offset) if days_offset is not None else None
+    sanitized_json = sanitize(unsanitized_json, datetime_offset)
+
+    # Encrypt the raw JSON file, if the user has indicated that this is desired by using either of
+    # the `--encrypt...` flags.
+    encryptor = get_encryptor_from_flags(encrypt_with, encrypt_with_gcp_kms)
+
+    # If no `encryptor` was passed in, this is an unencrypted write, so we can just dump the JSON
+    # into the `dest` file directly.
+    if encryptor is None:
+        dest.write(orjson.dumps(sanitized_json, option=orjson.OPT_INDENT_2 | orjson.OPT_UTC_Z))
```

Comment:
Is orjson specifically used since it is very fast?

Reply:
2 reasons:
1. We seem to be moving to `orjson` in other places anyway, so wanted to get ahead of that.
2. The default JSON serializer was tripping over one of the more complex classes, whereas `orjson` handled it fine.